content/sensu-go/5.17/guides/secrets-management.md

```diff
@@ -0,0 +1,289 @@
+Then, run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: VaultProvider
+api_version: secrets/v1
+metadata:
+  name: vault
+spec:
+  client:
+    address: http://localhost:8200
+    token: ROOT_TOKEN
+    version: v2
+    tls: null
+    max_retries: 2
+    timeout: 20s
+    rate_limiter:
+      limit: 10
+      burst: 100
+EOF
+{{< /highlight >}}
+
+### Create your secret
+
+_**NOTE**: Because you aren't using TLS, you will need to add `export VAULT_ADDR=http://127.0.0.1:8200` in your bash profile._
+
+First, retrieve your [PagerDuty Events API Integration Key][27].
+This is the secret you will set up in Vault.
+
+Next, open a new terminal and run `vault kv put secret/pagerduty key=INTEGRATION_KEY`.
+Replace `INTEGRATION_KEY` with your PagerDuty Events API Integration Key.
+
+This writes your secret into Vault.
+In this example, the name of the secret is `pagerduty`.
+The `pagerduty` secret contains a key, and you specified that the `key` value is your PagerDuty Events API Integration Key.
+The `id` value for your secret will be `secret/pagerduty#key`.
+
+The `id` value for secrets that target a HashiCorp Vault must start with the name of the secret's path in Vault.
+The Vault dev server is preconfigured with the `secret` keyspace already set up, so we recommend using the `secret/` path for the `id` value while you are learning and getting started with Vault secrets management.
+
+Run `vault kv get secret/pagerduty` to see the secret you just set up.
+
+Use `sensuctl create` to create your `vault` secret:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: pagerduty_key
+  namespace: default
+spec:
+  id: secret/pagerduty#key
+  provider: vault
+EOF
+{{< /highlight >}}
+
+Now you can securely pass your PagerDuty Events API Integration Key in Sensu checks, handlers, and mutators by referring to the `pagerduty_key` secret.
+In this guide, you'll use your `pagerduty_key` secret in a handler.
+
+## Add a handler
+
+### Register the PagerDuty Handler asset
+
+To begin, register the [Sensu PagerDuty Handler asset][23] with [`sensuctl asset add`][22]:
+
+{{< highlight shell >}}
+sensuctl asset add sensu/sensu-pagerduty-handler:1.2.0 -r pagerduty-handler
+{{< /highlight >}}
+
+This example uses the `-r` (rename) flag to specify a shorter name for the asset: `pagerduty-handler`.
+
+_**NOTE**: You can [adjust the asset definition][26] according to your Sensu configuration if needed._
+
+Run `sensuctl asset list --format yaml` to confirm that the asset is ready to use.
+
+With this handler, Sensu can trigger and resolve PagerDuty incidents.
+However, you still need to add your secret to the handler spec so that it requires your backend to request secrets from your secrets provider.
+
+### Add your secret to the handler spec
+
+To create a handler definition using Sensu's built-in `Env` secret provider, run:
```

Comment:
The handler definition will be the same regardless of the secret provider.

Reply:
Wouldn't the `secrets.secret` value be different? `env` for Env or `vault` for VaultProvider?

Reply:
Nope.
- secrets.Name is the name of the value you use in the command.. for example, I added `--dedup-key $DEDUP_KEY` to the example command.
- secrets.Secret is the name of the Sensu secret. In your example you named both Sensu secrets `pagerduty_key`.

Reply:
Thank you -- it made sense once I read on to your correction in the handler example.